src/python/pants/engine/http.py

```diff
@@ -0,0 +1,24 @@
+# Copyright 2019 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from dataclasses import dataclass
+from typing import Optional
+
+from pants.engine.rules import RootRule
+
+
+@dataclass(frozen=True)
+class HttpResponse:
+  response_code: Optional[int]
+  output_bytes: Optional[bytes]
+
+
+@dataclass(frozen=True)
+class MakeHttpRequest:
+  url: str
```

Comment:
Could we add a docstring to make it clear that to make an https request (as opposed to an http request) you need only to make the url have "https://" and don't need to specify anything else?

Reply:
I should test that making https requests with this API actually works, I'm not sure if there's any wierdness you have to do with rust reqwests to make that happen.